firestore/scripts/FriendlyEats.View.js

```diff
@@ -312,12 +342,33 @@ FriendlyEats.prototype.viewRestaurant = function(id) {
   return this.getRestaurant(id)
     .then(function(doc) {
       var data = doc.data();
+      if (that.userFavorites.indexOf(id) >= 0) {
+        console.log('This is a favorite restaurant');
+        data['isFavorite'] = true;
+      }
+
+      // This seems really old-school. Wonder if there's a better way
```

Comment:
You could read the user's favorites on this page, instead of relying on a URL param. That way, if someone was sent a link to the restaurant's page from a friend, you could still show that it is the user's favorite. Maybe a callable function that takes (userID, restaurantId) and returns true or false after checking whether the restaurant is a user's favorite?

Reply:
So, we're doing two related-but-different tasks here: The first thing is figuring out if this restaurant is one of our user's favorites, and that's something we do in lines 345-347, based on reading in the user doc at startup. We use that logic to determine what the logic of the "Add to favorites / remove from favorites" button should be. 

The second step is figuring out if we originally got to this page from the "main" page, or from the "favorites" page, and that's what lines 351-355 are doing. We use this logic to determine where we should take you when you click the close button. We need to do this separately, because even if a restaurant is your favorite, we could still get to this page from either location. That's kinda why I'm doing it through URL parameters.

The thing I realized after doing this, however, is that my URL parameters trick doesn't work if you visit a page from your favorites, then interact with other elements on the page -- it alters the URL and the argument doesn't carry over. So rather than keep trying to propagate this URL argument everywhere, I think the easiest solution is just to include a global "fromFavorites" variable (the same way we're storing the filters variable) which we can change in the viewList screen. (Hacky? Maybe, but it works). So that's what I've done in the latest iteration.